owncloudData/src/main/java/com/owncloud/android/data/authentication/datasources/implementation/OCLocalAuthenticationDataSource.kt

```diff
@@ -0,0 +1,245 @@
+     */
+    private fun addAccount(
+        lastPermanentLocation: String?,
+        serverInfo: ServerInfo,
+        userName: String,
+        password: String = "",
+        updateIfAlreadyExists: Boolean = false
+    ): Account {
+        if (lastPermanentLocation != null) {
+            serverInfo.baseUrl = WebdavUtils.trimWebdavSuffix(lastPermanentLocation)
+        }
+
+        val uri = Uri.parse(serverInfo.baseUrl)
+
+        val accountName = AccountUtils.buildAccountName(uri, userName)
+
+        if (accountExists(accountName)) {
+            if (!updateIfAlreadyExists) {
+                Timber.d("The account already exists")
+                throw AccountNotNewException()
+            } else {
+                return getAccountByName(accountName)!! // Account won't be null, since we've already checked it exists
+            }
+        } else {
+            val newAccount = Account(accountName, accountType)
+
+            // with external authorizations, the password is never input in the app
+            accountManager.addAccountExplicitly(newAccount, password, null)
+
+            /// add the new account as default in preferences, if there is none already
+            val defaultAccount: Account? = getCurrentAccount()
+            if (defaultAccount == null) {
+                val editor = PreferenceManager
+                    .getDefaultSharedPreferences(context).edit()
+                editor.putString("select_oc_account", accountName)
+                editor.apply()
+            }
+
+            return newAccount
+        }
+    }
+
+    private fun updateUserAndServerInfo(
+        newAccount: Account,
+        serverInfo: ServerInfo,
+        userInfo: UserInfo?,
+        userName: String
+    ) {
+        // include account version with the new account
+        accountManager.setUserData(
+            newAccount,
+            KEY_OC_ACCOUNT_VERSION,
+            ACCOUNT_VERSION.toString()
+        )
+
+        accountManager.setUserData(
+            newAccount, KEY_OC_VERSION, serverInfo.ownCloudVersion
+        )
+
+        accountManager.setUserData(
+            newAccount, KEY_OC_BASE_URL, serverInfo.baseUrl
+        )
+
+        if (userInfo != null) {
+            accountManager.setUserData(
+                newAccount, KEY_DISPLAY_NAME, userInfo.displayName
+            )
+        } else {
+            Timber.w("Couldn't get display name for %s", userName)
+        }
+    }
+
+    private fun accountExists(accountName: String?): Boolean {
+        val ocAccounts: Array<Account> = getAccounts()
+
+        if (accountName != null) {
+            var lastAtPos = accountName.lastIndexOf("@")
+            val hostAndPort = accountName.substring(lastAtPos + 1)
+            val username = accountName.substring(0, lastAtPos)
+            var otherHostAndPort: String
+            var otherUsername: String
+            val currentLocale: Locale = context.resources.configuration.locale
```

Comment:
[Here](https://stackoverflow.com/questions/14389349/android-get-current-locale-not-default) you have a proper explanation

Reply:
But it is actually deprecated. I would go with `Locale.getDefault()` or even `getLocales().get(0)` instead of the deprecated one.

Reply:
If I use `Locale.getDefault()` it can give us some problems if the user changes the device locale. 

About using `getLocales().get(0)`, see image below

![Screenshot 2020-03-16 at 12 54 01](https://user-images.githubusercontent.com/15261363/76756018-3bcfb200-6785-11ea-8594-e46606b3666d.png)